x/scheduler/keeper/keeper.go

```diff
@@ -21,80 +22,78 @@ import (
 )
 
 type Keeper struct {
-	cdc        codec.BinaryCodec
-	storeKey   storetypes.StoreKey
-	memKey     storetypes.StoreKey
-	paramstore paramtypes.Subspace
+	cdc      codec.BinaryCodec
+	storeKey store.KVStoreService
+	memKey   storetypes.StoreKey
 
 	account   types.AccountKeeper
 	EvmKeeper types.EvmKeeper
 
 	ider keeperutil.IDGenerator
 
-	Chains map[xchain.Type]xchain.Bridge
+	Chains    map[xchain.Type]xchain.Bridge
+	authority string
 }
 
 func NewKeeper(
 	cdc codec.BinaryCodec,
-	storeKey,
+	storeKey store.KVStoreService,
 	memKey storetypes.StoreKey,
-	ps paramtypes.Subspace,
 	account types.AccountKeeper,
 	evmKeeper types.EvmKeeper,
 	chains []xchain.Bridge,
+	authority string,
 ) *Keeper {
-	// set KeyTable if it has not already been set
-	if !ps.HasKeyTable() {
-		ps = ps.WithKeyTable(types.ParamKeyTable())
-	}
 
 	cm := slice.MustMakeMapKeys(chains, func(c xchain.Bridge) xchain.Type {
 		return c.XChainType()
 	})
 
 	k := &Keeper{
-		cdc:        cdc,
-		storeKey:   storeKey,
-		memKey:     memKey,
-		paramstore: ps,
-		account:    account,
-		EvmKeeper:  evmKeeper,
-		Chains:     cm,
+		cdc:       cdc,
+		storeKey:  storeKey,
+		memKey:    memKey,
+		account:   account,
+		EvmKeeper: evmKeeper,
+		Chains:    cm,
+		authority: authority,
 	}
 
 	k.ider = keeperutil.NewIDGenerator(k, nil)
 
 	return k
 }
 
-func (k Keeper) GetAccount(ctx sdk.Context, addr sdk.AccAddress) authtypes.AccountI {
+func (k Keeper) GetAccount(ctx context.Context, addr sdk.AccAddress) authtypes.AccountI {
 	return k.account.GetAccount(ctx, addr)
 }
 
-func (k Keeper) Logger(ctx sdk.Context) log.Logger {
-	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
+func (k Keeper) Logger(ctx context.Context) log.Logger {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	return liblog.FromSDKLogger(sdkCtx.Logger()).With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
 
 func (k Keeper) ModuleName() string {
 	return types.ModuleName
 }
 
-func (k Keeper) PreJobExecution(ctx sdk.Context, job *types.Job) error {
+func (k Keeper) PreJobExecution(ctx context.Context, job *types.Job) error {
 	return k.EvmKeeper.PreJobExecution(ctx, job)
 }
 
 // store returns default store for this keeper!
-func (k Keeper) Store(ctx sdk.Context) sdk.KVStore {
-	return ctx.KVStore(k.storeKey)
+func (k Keeper) Store(ctx context.Context) storetypes.KVStore {
```

Comment:
This is not correct. You are returning `memStore`, it should be kvstore.

Reply:
I pushed above fix with reference link with comment [here](https://github.com/palomachain/paloma/pull/1044/commits/7f8e981e399c3b70a0631685787434721c9dd1ae) Once verify if its correct usage @aleem1314  @byte-bandit